### PR TITLE
Update to use __DEV__ constant

### PR DESCRIPTION
**Pull request by Rahmon**
Change from `process.env.NODE_ENV === 'production'` to `__DEV__`.

Closes: #1577

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -112,6 +112,9 @@
     "semi": true
   },
   "jest": {
+    "globals": {
+      "__DEV__": true
```
Comment (Andreyco) [inline package.json]: `boolean` here, not `true`

Comment (Rahmon) [reply package.json]: I'll fix this.

Comment (Rahmon) [reply package.json]: I've done a new PR: #1596

File: index.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 'use strict'
 
-if (process.env.NODE_ENV === 'production') {
-  module.exports = require('./formik.cjs.production.js');
-} else {
+if (__DEV__) {
```
Comment (TrySound) [inline index.js]: Will this file appear in node_modules as is?

Comment (TrySound) [reply index.js]: /cc @jaredpalmer

Comment (Andreyco) [reply index.js]: it is replaced by `process.env.NODE_ENV === 'production'`

But as you can see, it is too verbose to type in every time

Comment (TrySound) [reply index.js]: Ah, ok.

**Pull request merged**